src/lib/Libnet/hnls.c

```diff
@@ -179,36 +179,65 @@ get_if_hostnames(struct sockaddr *saddr)
 	struct sockaddr_in6 *saddr_in6;
 	char buf[INET6_ADDRSTRLEN];
 	const char *bufp = NULL;
+#ifdef WIN32
+	char host[NI_MAXHOST] = {'\0'};
+	int ret = 0;	
+#endif /* WIN32 */
 
 	if (!saddr)
 		return NULL;
 
 	switch (saddr->sa_family) {
 		case AF_INET:
 			saddr_in = (struct sockaddr_in *)saddr;
+#ifdef WIN32
+			saddr_in->sin_family = AF_INET;
+#endif /* WIN32 */
 			addr = &saddr_in->sin_addr;
 			addr_size = sizeof(saddr_in->sin_addr);
 			bufp = inet_ntop(AF_INET, addr, buf, INET_ADDRSTRLEN);
 			if (!bufp)
 				return NULL;
+#ifdef WIN32
+			ret = getnameinfo((struct sockaddr_in *)saddr_in, sizeof(struct sockaddr_in), host, NI_MAXHOST, NULL, NULL, 0);
+			if (ret != 0 || !strlen(host))
```

Comment:
Instead of calling strlen(), just check host[0] != '\0'.  Saves us from traversing a string.